src/core/ddsc/tests/config.c

```diff
@@ -118,3 +118,423 @@ CU_Test(ddsc_config, incorrect_config, .init = ddsrt_init, .fini = ddsrt_fini) {
          "</"DDS_PROJECT_NAME">") == DDS_RETCODE_OK);
     CU_ASSERT_FATAL(dds_create_domain(2, "") == DDS_RETCODE_PRECONDITION_NOT_MET);
 }
+
+static bool patternmatch(const char* pattern, const char* string)
+{
+    char patterncharacter;
+
+    /* iterate over pattern string */
+    for (;;) {
+        patterncharacter = *pattern;
+        pattern++;
+        switch (patterncharacter) {
+        case '\0':
+            return (*string == '\0');
+        case '?':
+            if (*string == '\0') {
+                return false;
+            }
+            ++string;
+            break;
+        case '*':
+            if (*pattern == '\0'){
+                return true;
+            }
+            while (*string != '\0') {
+                if (patternmatch(pattern, string)) {
+                    return true;
+                }
+                ++string;
+            }
+            return false;
+            break;
+        default: /* Regular character */
+            if (*string != patterncharacter) {
+                return false;
+            }
+            string++;
+            break;
+        }
+    }
+}
+
+/*
+ * The 'found' variable will contain flags related to the expected log
+ * messages that were received.
+ * Using flags will allow to show that when message isn't received,
+ * which one it was.
+ */
+static uint32_t found;
+static void logger(void *ptr, const dds_log_data_t *data)
+{
+    char **expected = (char**)ptr;
+    for (uint32_t i = 0; expected[i] != NULL; i++) {
+        if (patternmatch(expected[i], data->message)) {
+            found |= (uint32_t)(1 << i);
+        }
+    }
+}
+
+CU_Test(ddsc_config, security_non, .init = ddsrt_init, .fini = ddsrt_fini) {
+
+    /* There shouldn't be traces that mention security. */
+    const char *log_expected[] = {
+      "*Security*",
+      NULL
+    };
+
+    dds_entity_t participant;
+
+    /* Set up the trace sinks to detect the config parsing. */
+    dds_set_log_mask(DDS_LC_FATAL|DDS_LC_ERROR|DDS_LC_WARNING|DDS_LC_CONFIG);
+    dds_set_log_sink(&logger, (void*)log_expected);
+    dds_set_trace_sink(&logger, (void*)log_expected);
+
+    /* Create participant with an empty security element. */
+    found = 0;
+    ddsrt_setenv(URI_VARIABLE, "<Tracing><Verbosity>finest</></>");
```

Comment:
Recently, a function named ``dds_create_domain`` was added that makes it possible to explicitly create a domain entity with the configuration passed in as a parameter. So if you use that one rather than ``dds_create_participant`` there is no longer a need to mess around with ``ddsrt_setenv``.

``dds_create_domain`` returns a handle that you can pass to ``dds_delete`` so the change is pretty trivial.

Reply:
> `dds_create_domain` returns a handle that you can pass to `dds_delete`.

This is unfortunately not true (yet).